configure.ac

```diff
@@ -1254,40 +1254,12 @@ dnl ncurses takes precedence.
 dnl
 AC_CHECK_HEADERS([curses.h curses/curses.h ncurses.h ncurses/ncurses.h])
 
-dnl Although n-library is preferred, only look for it if the n-header was found.
-CURSESLIBS=''
-PC_NAME_CURSES=""
-PC_LIBS_CURSES=""
-AS_IF([test x"$ac_cv_header_ncurses_h" = x"yes"], [
-    AC_CHECK_LIB(ncurses, printw,
-                 [AC_DEFINE(HAVE_LIBNCURSES,1, have ncurses library)])
-    CURSESLIBS=`$PKG_CONFIG --libs ncurses` || CURSESLIBS='-lncurses'
-    PC_NAME_CURSES="ncurses"
-])
-
-AS_IF([test x"$ac_cv_header_ncurses_ncurses_h" = x"yes"], [
-    AC_CHECK_LIB(ncurses, printw,
-                 [AC_DEFINE(HAVE_LIBNCURSES,1, have ncurses library)])
-    CURSESLIBS=`$PKG_CONFIG --libs ncurses` || CURSESLIBS='-lncurses'
-    PC_NAME_CURSES="ncurses"
-])
-
-dnl Only look for non-n-library if there was no n-library.
-AS_IF([test x"$CURSESLIBS" = x"" && test x"$ac_cv_header_curses_h" = x"yes"], [
-    AC_CHECK_LIB(curses, printw,
-                 [CURSESLIBS='-lcurses'; AC_DEFINE(HAVE_LIBCURSES,1, have curses library)])
-    PC_LIBS_CURSES="$CURSESLIBS"
-])
-
-dnl Only look for non-n-library if there was no n-library.
-AS_IF([test x"$CURSESLIBS" = x"" && test x"$ac_cv_header_curses_curses_h" = x"yes"], [
-    AC_CHECK_LIB(curses, printw,
-                 [CURSESLIBS='-lcurses'; AC_DEFINE(HAVE_LIBCURSES,1, have curses library)])
-    PC_LIBS_CURSES="$CURSESLIBS"
-])
-
-AS_IF([test x"$CURSESLIBS" != x""],
-      [PCMK_FEATURES="$PCMK_FEATURES ncurses"])
+save_LIBS=$LIBS
+AC_SEARCH_LIBS([printw], [ncurses curses],
+               [AC_DEFINE(HAVE_LIBNCURSES, 1, have ncurses library)
```

Comment:
This works, but only because we check for `defined(HAVE_LIBNCURSES) || defined(HAVE_LIBCURSES)` together.

We could move some of the detection mess in crm_mon.h to configure.ac. In particular if we don't have a compatible printw() we should be able to disable support in configure.ac rather than deal with it on the C end. We could define `CURSES_ENABLED` (renamed to `PCMK__ENABLE_CURSES`) directly instead of all the `HAVE_LIB` stuff, though I'm not sure if there's a way to specify which header to include without `HAVE`. We could at least ignore curses if we don't have any header.